Mathlib/Probability/Moments/CovarianceBilin.lean

```diff
@@ -1,3 +1,209 @@
+    covarianceBilin (μ.map (fun x ↦ c + x)) = covarianceBilin μ := by
+  by_cases h : MemLp id 2 μ
+  · ext x y
+    have h_Lp : MemLp id 2 (μ.map (fun x ↦ c + x)) :=
+      (measurableEmbedding_addLeft _).memLp_map_measure_iff.mpr <| (memLp_const c).add h
+    rw [covarianceBilin_apply h_Lp,
+      covarianceBilin_apply h, integral_map (by fun_prop) (by fun_prop)]
+    congr with z
+    rw [integral_map (by fun_prop) h_Lp.1]
+    simp only [id_eq]
+    rw [integral_add (integrable_const _)]
+    · simp
+    · exact h.integrable (by simp)
+  · ext
+    rw [covarianceBilin_of_not_memLp, covarianceBilin_of_not_memLp h]
+    rw [(measurableEmbedding_addLeft _).memLp_map_measure_iff.not]
+    contrapose! h
+    convert (memLp_const (-c)).add h
+    ext; simp
+
+lemma covarianceBilin_apply_basisFun {ι Ω : Type*} [Fintype ι] {mΩ : MeasurableSpace Ω}
+    {μ : Measure Ω} [IsFiniteMeasure μ] {X : ι → Ω → ℝ} (hX : ∀ i, MemLp (X i) 2 μ) (i j : ι) :
+    covarianceBilin (μ.map (fun ω ↦ toLp 2 (X · ω)))
+      (basisFun ι ℝ i) (basisFun ι ℝ j) = cov[X i, X j; μ] := by
+  have (i : ι) := (hX i).aemeasurable
+  rw [covarianceBilin_apply_eq_cov, covariance_map]
+  · simp [basisFun_inner]; rfl
+  · exact Measurable.aestronglyMeasurable (by fun_prop)
+  · exact Measurable.aestronglyMeasurable (by fun_prop)
+  · fun_prop
+  · exact (memLp_map_measure_iff aestronglyMeasurable_id (by fun_prop)).2 (MemLp.of_eval_piLp hX)
+
+lemma covarianceBilin_apply_basisFun_self {ι Ω : Type*} [Fintype ι] {mΩ : MeasurableSpace Ω}
+    {μ : Measure Ω} [IsFiniteMeasure μ] {X : ι → Ω → ℝ} (hX : ∀ i, MemLp (X i) 2 μ) (i : ι) :
+    covarianceBilin (μ.map (fun ω ↦ toLp 2 (X · ω)))
+      (basisFun ι ℝ i) (basisFun ι ℝ i) = Var[X i; μ] := by
+  rw [covarianceBilin_apply_basisFun hX, covariance_self]
+  have (i : ι) := (hX i).aemeasurable
+  fun_prop
+
+lemma covarianceBilin_apply_pi {ι Ω : Type*} [Fintype ι] {mΩ : MeasurableSpace Ω}
+    {μ : Measure Ω} [IsFiniteMeasure μ] {X : ι → Ω → ℝ}
+    (hX : ∀ i, MemLp (X i) 2 μ) (x y : EuclideanSpace ℝ ι) :
+    covarianceBilin (μ.map (fun ω ↦ toLp 2 (X · ω))) x y =
+      ∑ i, ∑ j, x i * y j * cov[X i, X j; μ] := by
+  have (i : ι) := (hX i).aemeasurable
+  nth_rw 1 [covarianceBilin_apply_eq_cov, covariance_map_fun, ← (basisFun ι ℝ).sum_repr' x,
+    ← (basisFun ι ℝ).sum_repr' y]
+  · simp_rw [sum_inner, real_inner_smul_left, basisFun_inner, PiLp.toLp_apply]
+    rw [covariance_fun_sum_fun_sum]
+    · refine Finset.sum_congr rfl fun i _ ↦ Finset.sum_congr rfl fun j _ ↦ ?_
+      rw [covariance_mul_left, covariance_mul_right]
+      ring
+    all_goals exact fun i ↦ (hX i).const_mul _
+  any_goals exact Measurable.aestronglyMeasurable (by fun_prop)
+  · fun_prop
+  · exact (memLp_map_measure_iff aestronglyMeasurable_id (by fun_prop)).2 (MemLp.of_eval_piLp hX)
+
+section covarianceOperator
+
+variable [CompleteSpace E]
+
+/-- The covariance operator of the measure `μ`. -/
```

Comment:
Could you recall mathematically what this is? (There is a subtlety here that you should insist on, since you're using `uncenteredCovarianceBilinDual`)